engine/time/src/main/java/io/deephaven/time/calendar/BusinessCalendar.java

```diff
@@ -1,774 +1,1898 @@
 /**
- * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
  */
 package io.deephaven.time.calendar;
 
-import java.time.DayOfWeek;
-import java.time.Instant;
-import java.time.LocalDate;
-import java.util.List;
-import java.util.Map;
+import io.deephaven.base.verify.Require;
+import io.deephaven.base.verify.RequirementFailure;
+import io.deephaven.time.DateTimeUtils;
+
+import java.time.*;
+import java.util.*;
 
 /**
- * A business calendar. Calendar is extended with the concept of business and non-business time.
- *
- * To comply with the ISO-8601 standard for dates, Strings should be of the form "yyyy-MM-dd",
+ * A business calendar, with the concept of business and non-business time.
+ * <p>
+ * Date strings must be in a format that can be parsed by {@code DateTimeUtils#parseDate}. Methods that accept strings
+ * can be slower than methods written explicitly for {@code Instant}, {@code ZonedDateTime}, or {@code LocalDate}.
  */
-public interface BusinessCalendar extends Calendar {
+public class BusinessCalendar extends Calendar {
```

Comment:
It was originally an interface, but it seemed Rube Goldberg for this case since I don't think there is an expectation that others create business calendars that are not based on `BusinessCalendar`.  Do you still want an interface?